snap/hooks/post-refresh

```diff
@@ -5,3 +5,6 @@ if [ ! -f "$SNAP_DATA/config/daemon.json" ]; then
     mkdir -p "$SNAP_DATA/config"
     cp "$SNAP/config/daemon.json" "$SNAP_DATA/config/daemon.json"
 fi
+
+# Should should always be re-craeted if required, as it contains paths based on snap revision #
+rm -f "${SNAP_DATA}/etc/cdi/nvidia.yaml"
```

Comment:
snap revisions of what? docker itself? but SNAP_DATA is per-snap-revision directory already. Or does it get copied up between refreshes?

Reply:
An example path from the CDI config:
```
/snap/docker/2893/usr/bin/nvidia-ctk
```

As the CDI config file is in `$SNAP_DATA`, it get's copied on snap update.  It does need to be generated in-situ on the host.

I'm not aware of a better way to handle this scenario, and this seems reasonable.

Reply:
The rest of the configs are in $SNAP_DATA, and you would be able to do it in snap data too.... for example I would expect it to use `/snap/docker/current/usr/bin/nvidia-ctk` or even just use `nvidia-ctk` as that path should be in `$PATH` and just work.

I'll play with this, but I don't like generating/regenerating files with snap-internal paths, when it really ought to be docker revision agnostic, especially when none of the other nvidia things change.

Reply:
That path used for that command is specified when we generate the config.  We can change it from:
```
"${SNAP}/usr/bin/nvidia-ctk" cdi generate --nvidia-ctk-path "${SNAP}/usr/bin/nvidia-ctk" --output="${SNAP_DATA}/etc/cdi/nvidia.yaml"
```
to
```
"${SNAP}/usr/bin/nvidia-ctk" cdi generate --nvidia-ctk-path "/snap/SNAP_NAME/current/usr/bin/nvidia-ctk" --output="${SNAP_DATA}/etc/cdi/nvidia.yaml"
```
I will update and remove the purge.

I tested no fully qualified path, but the canonical path is then discovered and used [ include snap revision ].

Reply:
Updated - PTAL